File: chat-sdk/src/main/java/ru/usedesk/chat_sdk/external/entity/UsedeskSingleLifeEvent.java
package ru.usedesk.chat_sdk.external.entity;

import androidx.annotation.Nullable;

public class UsedeskSingleLifeEvent<DATA> extends UsedeskEvent<DATA> {
    private boolean processed = false;

    public UsedeskSingleLifeEvent() {

    }

    public UsedeskSingleLifeEvent(@Nullable DATA data) {
        super(data);
    }

    @Override
    public boolean isProcessed() {
        return processed;
    }

    @Override
    public void setProcessed() {
        this.processed = true;
    }
}

File: chat-sdk/src/main/java/ru/usedesk/chat_sdk/external/entity/UsedeskSingleLifeEvent.kt
package ru.usedesk.chat_sdk.external.entity

class UsedeskSingleLifeEvent<DATA>(
        data: DATA
) : UsedeskEvent<DATA>(data) {
    private var isProcessed = false

    override fun process(onProcess: (DATA) -> Unit) {
        if (!isProcessed) {
            isProcessed = true
            super.process(onProcess)
        }
    }
}